src/Catty/ViewController/Continue&New/SceneTableViewController.m

```diff
@@ -147,12 +114,47 @@ - (void)addObjectActionWithName:(NSString*)objectName
     ltvc.showAddLookActionSheetAtStartForScriptEditor = NO;
     ltvc.afterSafeBlock =  ^(Look* look) {
         [self.navigationController popViewControllerAnimated:YES];
+        NSString *initialText = @"";
+        if (look) {
+            if (![look.name  isEqual: @"look"]) {
+                initialText = look.name;
+            }
+            [[[[[[[[AlertControllerBuilder textFieldAlertWithTitle:kLocalizedAddObject message:[NSString stringWithFormat:@"%@:", kLocalizedObjectName]]
+                  placeholder:kLocalizedEnterYourObjectNameHere]
+                initialText:initialText]
+                 addCancelActionWithTitle:kLocalizedCancel handler:^{
+                NSUInteger index = kBackgroundObjectIndex + indexPath.section + indexPath.row;
+                SpriteObject *object = (SpriteObject*)[self.scene.objects objectAtIndex:index];
+                [self.scene removeObject:object];
+                [self.scene.project saveToDiskWithNotification:NO];
+                [self.tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:((indexPath.row != 0) ? UITableViewRowAnimationTop : UITableViewRowAnimationFade)];
+                [self cancelAddingObjectFromScriptEditor];
+            }]
+                addDefaultActionWithTitle:kLocalizedOK handler:^(NSString *name) {
+                [self renameObjectActionToName:name spriteObject:(SpriteObject*)[self.scene.objects objectAtIndex:(kBackgroundObjectIndex + indexPath.section + indexPath.row)]];
+            }]
+               valueValidator:^InputValidationResult *(NSString *name) {
+                InputValidationResult *result = [Util validationResultWithName:name
+                                                                     minLength:kMinNumOfObjectNameCharacters
+                                                                     maxlength:kMaxNumOfObjectNameCharacters];
+                if (!result.valid) {
+                    return result;
+                }
+                // Alert for Objects with same name
+                if ([[self.scene allObjectNames] containsObject:name]) {
```

Comment:
This doesn't seem to work, when trying to save an image with the same name, it just gets assigned a unique name (e.g. "Look (1)")